core/primitives/src/views.rs

```diff
@@ -1019,7 +1019,6 @@ pub struct ExecutionOutcomeView {
     /// Execution status. Contains the result in case of successful execution.
     pub status: ExecutionStatusView,
     /// Execution metadata, versioned
-    #[serde(skip)]
     pub metadata: ExecutionMetadata,
```

Comment:
Done. Note that the key is `Cost` instead of `String` to serde serialialize. As a result it need to be Vec<(Cost, u64)> instead of BTreeMap as serde-json doesn't accept nested enum as json keys. And we do not need any lookup from the view so a vector of key value pair just works. An example output:
```
"costs": [[{"ExtCost": "base"}, 1323840555], [{"ExtCost": "contract_compile_base"}, 35445963], ...
```

Reply:
```
[[{"ExtCost": "base"}, 1323840555], [{"ExtCost": "contract_compile_base"}, 35445963]
```

1. This is not a self-explanatory format at all; I am completely lost here
2. Please, avoid using the default externally tagged enum representations since it is quite inconvenient format for the client code. Internally tagged or adjacently tagged representations are much easier to work with. More details about enum representations: https://serde.rs/enum-representations.html
3. Please, don't expose unnamed fields (`[{}, 1323840555]`)

Reply:
>  This is not a self-explanatory format at all; I am completely lost here
    Please, avoid using the default externally tagged enum representations since it is quite inconvenient format for the client code. Internally tagged or adjacently tagged representations are much easier to work with. More details about enum representations: https://serde.rs/enum-representations.html

This is rather a personal thing. I do have read this document before this design. 

If you are mean to replace enum part, from externally to internally tagged repr, such as from `{"ExtCost": "base"}` to `{"category":"ExtCost", "kind":"base"}`, I do think the previous one is more clear, since later you also need to document, to document what a category means.
If you mean the entire costs, It does not fit the model since costs is a BTreeMap<Enum, u64>, the tagged or adjacently tagged enum repr for enum is for the enum itself. In json, the resulting enum `{"ExtCost": "base"}` cannot be a map key, therefore, it's natural for the key-value map listed as array. 

> Please, don't expose unnamed fields ([{}, 1323840555])

This does not happen